### DISCUSSION TITLE
Which tracking algorithm is the best one for fast moving objects?

### DISCUSSION THREAD

**Discussion by ozaanme** [mikel-brostrom/yolo_tracking#886] [Q&A]
I use yolov8 for ball detection in table tennis and I tried to use StrongSORT to evaluate each shot separately. Any idea which of the SORT algorithms is best for tracking fast moving objects? Or how can I increase the performance by changing which parameters?

**Comment by mikel-brostrom** [comment]
I guess you are experimenting undershooting as you are working with fast moving objects. What is relevant here is not the tracking method but the tuning in the KFs in each method. I would recommend you to read some KF theory. But in short, this can potentially be solved by adjusting a combination of different values:

- Noise covariance matrix (Q): The process noise covariance matrix represents the uncertainty in the system dynamics. Increasing the values in this matrix can reduce overshooting, while decreasing them can reduce undershooting. 
- Noise covariance matrix (R): The measurement noise covariance matrix represents the system's uncertaint. Similar to the process noise matrix, adjusting the values in this matrix can help mitigate overshooting or undershooting.
- Kalman gain (K): The Kalman gain determines the weight given to the predicted state estimate and the measurement update. A higher Kalman gain emphasizes the measurement update and can help reduce overshooting, while a lower gain can reduce undershooting.
